### ISSUE TITLE
CD - cannot load imagenet pretrained weights (L2)

### ISSUE THREAD

**Issue by nikhilmakkar** [lwCVer/LWGANet#12]
There is no init_cfg file. No matter what I do it always start with scratch. 

<img width="356" alt="Image" src="https://github.com/user-attachments/assets/e6d422b7-7fff-4137-917f-1c8df53016a9" />

Changed the path directly here too 

<img width="542" alt="Image" src="https://github.com/user-attachments/assets/c36d3bca-13d9-4a00-8128-ac606e9de20c" />

Still always starting from scratch

**Comment by lwCVer** [comment]
> There is no init_cfg file. No matter what I do it always start with scratch.
> 
> <img alt="Image" width="356" src="https://private-user-images.githubusercontent.com/16943067/434761813-e6d422b7-7fff-4137-917f-1c8df53016a9.PNG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQ5NTMwMjEsIm5iZiI6MTc0NDk1MjcyMSwicGF0aCI6Ii8xNjk0MzA2Ny80MzQ3NjE4MTMtZTZkNDIyYjctN2ZmZi00MTM3LTkxN2YtMWM4ZGY1MzAxNmE5LlBORz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDE4VDA1MDUyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ5MmNhODU3MjcyNmQzMGVlNGZjZGQzZTNjM2NiODhiNGJmZjlkYTBjM2UwOWQxZGQ4MTg2MDQyMGY1ZmI2ZTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.JLVfAkriYHEJEUrBiDYIhTQ2LGhc1XiWbFd6gS9Fqb8">
> Changed the path directly here too
> 
> <img alt="Image" width="542" src="https://private-user-images.githubusercontent.com/16943067/434761936-c36d3bca-13d9-4a00-8128-ac606e9de20c.PNG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQ5NTMwMjEsIm5iZiI6MTc0NDk1MjcyMSwicGF0aCI6Ii8xNjk0MzA2Ny80MzQ3NjE5MzYtYzM2ZDNiY2EtMTNkOS00YTAwLTgxMjgtYWM2MDZlOWRlMjBjLlBORz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDE4VDA1MDUyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQxN2ZkOTQxYzJjY2RmN2JlYTBhMTcyMzYwYzkyOTZhZDVmZDdmN2M3ZTQ3Y2MyMDA5YzgyMjkwNmVkZjgwYTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.D9HCZHaRhnytzc8cTsrO4GegE2A3NN34somWFH6guRg">
> Still always starting from scratch

Thank you for your attention to the work and your positive feedback. We have updated the LWGANet in the A2Net and CLAFA codes. To put it simply, we have removed 'pretrained=pretrained' because the pre-trained weights were loaded in the judgment statement.

Regarding the mismatch in the "norm" layer among the missing keys, we have incorporated normalization in the backbone network for downstream tasks to enhance model robustness, which has minimal impact on final results. As for the "avgpool" and "head" errors in unexpected keys, these weights originate from the classification model and contain classification headers. However, since downstream tasks do not utilize these weight components, they do not affect model performance. If you find these messages visually distracting, you may comment out the print statements print('missing_keys: ', missing_keys) and print('unexpected_keys: ', unexpected_keys) in the backbone code.

**Comment by nikhilmakkar** [comment]
Thank you for your reply. I did not understand the first part. What you mean by 'because the pre-trained weights were loaded in the judgment statement.'?

I can see that you removed 'pretrained = pretrained' yesterday. I cloned 2-3 days ago so it still had those. I will remove that.

I am still confused though where the imagenet weights are being loaded or the model is being trained from scratch?

**Comment by lwCVer** [comment]
> Thank you for your reply. I did not understand the first part. What you mean by 'because the pre-trained weights were loaded in the judgment statement.'?
> 
> I can see that you removed 'pretrained = pretrained' yesterday. I cloned 2-3 days ago so it still had those. I will remove that.
> 
> I am still confused though where the imagenet weights are being loaded or the model is being trained from scratch?

For example:
In the following code snippet:

********************************
if pretrained == True:
    checkpoint = torch.load('./backbone_weights/lwganet_l2.pth',
                            map_location=torch.device('cuda:0'))
    _state_dict = checkpoint['model']
    state_dict = _state_dict
    missing_keys, unexpected_keys = model.load_state_dict(state_dict, False)

    # show for debug
    print('missing_keys: ', missing_keys)
    print('unexpected_keys: ', unexpected_keys)
********************************
The code within the if statement represents the weight loading process. When you correctly specify the backbone weight path, the network will properly load the backbone weights while keeping the decoder part trainable. 

We have updated LWGANet to include print() statements that explicitly indicate whether weights are successfully loaded, addressing your feedback. Thank you for your input. Wishing you a pleasant day!

**Comment by nikhilmakkar** [comment]
Thank you for the help. It is working now.!
